### DISCUSSION TITLE
Interpreting Coverage of SOUR Records

### DISCUSSION THREAD

**Discussion by Snake69** [FamilySearch/GEDCOM#467] [Q&A] [answered]
I'm developing genealogy software which will import data into a proprietary database layout.  One of the ways to import the data will be via Gedcom.

One requirement for importing an event is that it be sourced.  If not sourced, the event will be passed over.

With that in mind, it appears there will be instances where it is unclear (to me) the extent of the coverage of certain SOUR records.  One example is I have seen Gedcoms where there is a

0 INDI record followed by a
1 NAME record followed by a
2 SOUR record

The INDI section may contain many events but no other SOUR records.  Should I interpret the one 2 SOUR record to apply to all events?  And if the 0 FAM section pointed to by the 1 FAMS record in the 0 INDI record contains no SOUR records, should the one 2 SOUR in the INDI record cover the marriage and the children?

I've looked around and have found nothing covering this sort of thing.  Is this something where I'm just going to have to use my best judgement?  Or perhaps there's a guide or discussion someplace?

Marshall

**Comment by tychonievich** [comment, marked as answer]
The specification defines [`SOURCE_CITATION`](https://gedcom.io/specifications/FamilySearchGEDCOMv7.html#SOURCE_CITATION) as

> A citation indicating that the pointed-to source record supports the claims made in the superstructure. 

Thus, a `2 SOUR` describes a source of the level-1 structure it is contained in and the substructures thereof. In your example, that would mean it is the source of the `NAME` structure (and its substructures, if any).

Note that this model (retained from GEDCOM's earliest versions) is a bit less precise than we might hope. For example,

```gedcom
0 @I1@ INDI
1 NAME Jane /Bloggs/
1 BIRT
2 DATE 1 JAN 1970
1 DEAT
2 DATE AFT 1993
1 SOUR @S1@
```

has a source for the entire individual, without making it clear if that source provides all of the details in the `INDI` record or only a subset of them. We've have many conversations about possible more source-oriented structures over the year, but none have yet risen to a level of maturity and simplicity to make it into a proposed future GEDCOM version.
```

**Comment by Norwegian-Sardines** [comment]
I would not interpret a level 2 `SOUR` tag on a `NAME` tag as applied to all “facts” for the simple reason that there is available a level 1 `SOUR` tag.   My rule is that a `SOUR` tag only applies to the prior level.

That being said, it is possibly the sender is very lazy and did not want to repeat multiple links when they found all facts for the individual in one place, but I feel that you must set a rule.
